src/seedu/addressbook/data/person/Name.java

```
@@ -67,7 +67,10 @@ public int hashCode() {
      */
     public boolean isSimilar(Name other) {
         
-        if (this.fullName.toLowerCase().equals(((Name)other).fullName.toLowerCase())){
+        if (other == null){
+            return false;
```

You should throw an exception instead of returning false.